pkg/utils/node/predicates.go

```diff
@@ -31,7 +31,28 @@ func IsReadyAndSchedulable(node v1.Node) bool {
 	return false
 }
 
-func IsPastEmptyTTL(node *v1.Node) bool {
+func IsReady(node v1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady {
+			return condition.Status == v1.ConditionTrue
+		}
+	}
+	return false
+}
+
+func FailsToBecomeReady(node v1.Node) bool {
+	if time.Since(node.GetCreationTimestamp().Time) < (15 * time.Minute) {
```

Comment:
Didn't we test this to be 5 minutes before pods are rescheduled?

Reply:
It would be great if we could find the line of code upstream that does this (maybe in replicaset?)

Reply:
I'm noticing 
```
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
```
on my pods.

Reply:
It looks like even if we set this value, these tolerations are configurable [here](
https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions), meaning that we could have severely different logic depending on user configuration. 

While 15 minutes doesn't capture all cases, not leaving any room for leeway after the default timeout value could lead to some race conditions in terms of propagating pod state. It could be that setting a default value like this might not be the answer for a lot of users.

Reply:
Talked offline, will also check to see if all the pods on the node are terminating as well.